hack/test-e2e-batch-gateway.sh

```
@@ -0,0 +1,92 @@
+    fi
+
+    # Verify worker pool is active
+    pool_metric=$(get_async_metric "llm_d_async_async_pool_worker_limit")
+    if [ "$pool_metric" -gt 0 ]; then
+        log "Worker pool active: pool_worker_limit=$pool_metric"
+    else
+        die "Worker pool not active (async_pool_worker_limit=0)"
+    fi
+
+    # Verify gate budget if a gate type is configured
+    local gate_type
```

Shellcheck (0.11.0)

[error] 'local' is only valid in functions.

(SC2168)